packages/nodes-base/nodes/Redis/utils.ts

```diff
@@ -16,8 +16,17 @@ export function setupRedisClient(credentials: RedisCredential, isTest = false):
 		port: credentials.port,
 		tls: credentials.ssl === true,
 		connectTimeout: 10000,
-		// Disable reconnection for tests to prevent hanging
-		reconnectStrategy: isTest ? false : undefined,
+		reconnectStrategy: (retries: number, cause: NodeJS.ErrnoException) => {
+			// stop retrying immediately if socket timeout or after 10 retries
+			if (cause.code === 'ECONNREFUSED' || retries >= 10) {
+				return false;
+			}
+
+			const jitter = Math.floor(Math.random() * 200);
+			const delay = Math.min(Math.pow(2, retries) * 50, 2000);
+
+			return delay + jitter;
+		},
```

Comment:
Nicely done with the exponential backoff and jitter :tada:! I do think we should use a longer retry interval, the current code will retry for ~15s total, which might not cover service restarts

```ts
                reconnectStrategy: (retries: number, cause: NodeJS.ErrnoException) => {
                        // Retry for ~15min
                        if (retries >= 10) {
                                return false;
                        }

                        const jitter = Math.floor(Math.random() * 1000);
                        const delay = Math.pow(2, retries) * 1000;

                        return delay + jitter;
                },
                ```
```